src/Functions/FunctionsConversion.h

```diff
@@ -4561,7 +4561,7 @@ arguments, result_type, input_rows_count); \
 
                 if (from_low_cardinality)
                 {
-                    const auto * col_low_cardinality = typeid_cast<const ColumnLowCardinality *>(arguments[0].column.get());
+                    const auto * col_low_cardinality = assert_cast<const ColumnLowCardinality *>(arguments[0].column.get());
```

Comment:
Keep in mind that this is only going to work in debug mode. It's still going to crash in release

Reply:
Do you think it's better to add a separate check `if (col_low_cardinality) throw ...`?

Reply:
Ah, I see you found repro, this is great. Are you going to debug it? If no, I can do it

Reply:
I'm trying to fix other things right now, so if you are familiar with dictionaries or conversion functions you can have it. Otherwise I'll go through the fuzzer issues as soon as builds are fine

Reply:
I will take it then